### PR TITLE
Enabling more examples [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by jiajiechen**
* working on reuters_mlps_relu_vs_selu example
* fix code style and usage

@sandeep-krishnamurthy

### CODE REVIEW

File: keras/backend/mxnet_backend.py [hunk 1]
```diff
@@ -1849,7 +1857,12 @@ def greater(x, y):
         scalar = True
     if scalar:
         return KerasSymbol(mx.sym.Cast(x > y, dtype='uint8'))
-    return KerasSymbol(mx.sym.Cast(mx.sym.broadcast_greater(lhs=x, rhs=y), dtype='uint8'))
+    print("[Debug]", x)
+    print("[Debug]", y)
+    if isinstance(x, mx.sym.Symbol) and isinstance(y, mx.sym.Symbol):
+        return KerasSymbol(mx.sym.Cast(mx.sym.broadcast_greater(lhs=x, rhs=y), dtype='uint8'))
+    else:
+        raise TypeError
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: Please add a good message for these type error

File: keras/backend/mxnet_backend.py [hunk 2]
```diff
@@ -1076,6 +1026,7 @@ def random_uniform_variable(shape, low, high, dtype=None,
     return kvar
 
 
+# TODO: depreciated
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: Why are these marked deprecated? They do exist in latest Keras.

File: keras/backend/mxnet_backend.py [hunk 3]
```diff
@@ -1180,8 +1131,10 @@ def cast(x, dtype):
     if isinstance(x, KerasSymbol):
         return KerasSymbol(
             mx.sym.Cast(data=x.symbol, dtype=dtype))
-    else:
+    elif hasattr(x, astype):
         return x.astype(dtype)
+    else:
+        raise TypeError("The input is invalid for cast operation.")
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: Lets add this to all error message from MXNet backend - "MXNet Backend: .."

File: keras/backend/mxnet_backend.py [hunk 4]
```diff
@@ -1855,7 +1806,10 @@ def equal(x, y):
         scalar = True
     if scalar:
         return KerasSymbol(mx.sym.Cast(x == y, dtype='uint8'))
-    return KerasSymbol(mx.sym.Cast(mx.sym.broadcast_equal(lhs=x, rhs=y), dtype='uint8'))
+    if isinstance(x, mx.sym.Symbol) and isinstance(y, mx.sym.Symbol):
+        return KerasSymbol(mx.sym.Cast(mx.sym.broadcast_equal(lhs=x, rhs=y), dtype='uint8'))
+    else:
+        raise TypeError
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: Please add a descriptive message for the issue. Same in other places.

File: keras/backend/mxnet_backend.py [hunk 5]
```diff
@@ -1901,7 +1858,12 @@ def greater(x, y):
         scalar = True
     if scalar:
         return KerasSymbol(mx.sym.Cast(x > y, dtype='uint8'))
-    return KerasSymbol(mx.sym.Cast(mx.sym.broadcast_greater(lhs=x, rhs=y), dtype='uint8'))
+    print("[Debug]", x)
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: To be removed?

File: keras/backend/mxnet_backend.py [hunk 6]
```diff
@@ -1924,7 +1886,12 @@ def greater_equal(x, y):
         scalar = True
     if scalar:
         return KerasSymbol(mx.sym.Cast(x >= y, dtype='uint8'))
-    return KerasSymbol(mx.sym.Cast(mx.sym.broadcast_greater_equal(lhs=x, rhs=y), dtype='uint8'))
+    print("[Debug]", x)
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: removed?

File: keras/backend/mxnet_backend.py [hunk 7]
```diff
@@ -2820,20 +2796,22 @@ def in_train_phase(x, alt, training=None):
...
             return x
-    elif training == 0:
+    elif training is 0:
         if callable(alt):
             return alt()
         else:
             return alt
-    raise AssertionError("Learning phase must be 0 or 1.")
+
+    x = switch(training, x, alt)
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: Is this tested?

File: keras/backend/mxnet_backend.py [hunk 8]
```diff
@@ -3054,7 +3031,10 @@ def dropout(x, level, noise_shape=None, seed=None):
     if not 0 <= level <= 1:
         raise ValueError("MXNet Backend: Invalid level provided for dropout '{0}'. "
                          "Expected between 0 and 1.".format(level))
-    _seed_mxnet(seed)
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: I think it is good to keep _seed_mxnet() function.

File: keras/utils/test_utils.py [hunk 9]
```diff
@@ -88,6 +88,7 @@ def layer_test(layer_cls, kwargs={}, input_shape=None, input_dtype=None,
     # check shape inference
     model = Model(x, y)
     expected_output_shape = layer.compute_output_shape(input_shape)
+    model.compile('rmsprop', 'mse')
```
Comment (sandeep-krishnamurthy) [inline keras/utils/test_utils.py]: Will this change break other Backend like TF or CNTK?

File: tests/keras/backend/backend_test.py [hunk 10]
```diff
@@ -814,6 +812,7 @@ def test_conv2d(self):
             with pytest.raises(ValueError):
                 k.conv2d(k.variable(xval), k.variable(kernel_val), data_format='channels_middle')
 
+    @pytest.mark.skip
```
Comment (sandeep-krishnamurthy) [inline tests/keras/backend/backend_test.py]: As discussed please add message on why we skip this.
Also, This should be skipped only if K.backend() == 'mxnet'

File: keras/backend/mxnet_backend.py (hunk 7 above)
Comment (jiajiechen) [reply keras/backend/mxnet_backend.py]: yes

File: keras/backend/mxnet_backend.py (hunk 1 above)
Comment (jiajiechen) [reply keras/backend/mxnet_backend.py]: added.

File: keras/backend/mxnet_backend.py [hunk 11]
```diff
@@ -267,7 +208,7 @@ def __init__(self, mx_symbol, symbol_name=None, neighbors=None, is_var=False):
         else:
             self._train_sym = mx_symbol if learning_phase() else None
             self._pred_sym = None if learning_phase() else mx_symbol
-        self._name = symbol_name
```
[the remark this answers is not in the document]
Comment (jiajiechen) [reply keras/backend/mxnet_backend.py]: removed

File: keras/backend/mxnet_backend.py [hunk 12]
```diff
@@ -351,16 +292,17 @@ def __getitem__(self, in_slice):
...
                 end.append(i + 1)
-            else:
-                assert isinstance(i, slice)
+            elif isinstance(i, slice):
                 assert i.step is None or i.step == 1
                 begin.append(i.start)
                 end.append(i.stop)
-        return KerasSymbol(mx.sym.slice(self.symbol, begin=tuple(begin), end=tuple(end)), neighbor=[self])
+            else:
+                raise AttributeError
```
[the remark this answers is not in the document]
Comment (jiajiechen) [reply keras/backend/mxnet_backend.py]: added.

File: keras/utils/test_utils.py (hunk 9 above)
Comment (jiajiechen) [reply keras/utils/test_utils.py]: No.

**Pull request merged**